File: skills/faion/knowledge/pro/dev/software-developer/php-laravel-patterns/templates/controller.php
// purpose: Thin layered controller skeleton
// consumes: see content/02-output-contract.xml inputs
// produces: artefact conforming to content/02-output-contract.xml
// depends-on: content/01-core-rules.xml
// token-budget-impact: ~350 tokens when loaded as context

<?php

namespace App\Http\Controllers\Api\V1;

use App\Http\Controllers\Controller;
use App\Http\Requests\StoreOrderRequest;
use App\Http\Requests\UpdateOrderRequest;
use App\Http\Resources\OrderResource;
use App\Models\Order;
use App\Services\OrderService;
use Illuminate\Http\Resources\Json\AnonymousResourceCollection;

class OrderController extends Controller
{
    public function __construct(private readonly OrderService $service)
    {
        $this->authorizeResource(Order::class, 'order');
    }

    public function index(): AnonymousResourceCollection
    {
        return OrderResource::collection($this->service->listForCurrentUser());
    }

    public function store(StoreOrderRequest $request): OrderResource
    {
        return new OrderResource($this->service->create($request->validated()));
    }

    public function update(UpdateOrderRequest $request, Order $order): OrderResource
    {
        return new OrderResource($this->service->update($order, $request->validated()));
    }

    public function destroy(Order $order): void
    {
        $this->service->cancel($order);
    }
}

File: skills/faion/knowledge/pro/dev/software-developer/java-spring/templates/Controller.java
// purpose: thin @RestController skeleton conforming to thin-controller rule
// consumes: OrdersService + request DTO
// produces: HTTP controller artefact
// depends-on: content/01-core-rules.xml
// token-budget-impact: ~300 tokens when loaded as reference

package faion.web.orders;

import faion.app.orders.OrdersService;
import jakarta.validation.Valid;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.UUID;

@RestController
@RequestMapping("/api/orders")
public class OrdersController {

    private final OrdersService service;

    public OrdersController(OrdersService service) {
        this.service = service;
    }

    @GetMapping("/{id}")
    public ResponseEntity<OrderResponse> getById(@PathVariable UUID id) {
        return ResponseEntity.ok(service.getById(id));
    }

    @GetMapping
    public Page<OrderSummary> list(Pageable pageable) {
        return service.list(pageable);
    }

    @PostMapping
    public ResponseEntity<OrderResponse> create(@Valid @RequestBody CreateOrderRequest req) {
        OrderResponse created = service.create(req);
        return ResponseEntity.created(java.net.URI.create("/api/orders/" + created.id())).body(created);
    }
}
